pandas/core/arrays/base.py

```diff
@@ -1133,7 +1157,7 @@ def take(self, indices, allow_fill=False, fill_value=None):
         # pandas.api.extensions.take
         raise AbstractMethodError(self)
 
-    def copy(self: ExtensionArrayT) -> ExtensionArrayT:
+    def copy(self) -> ExtensionArray:
```

Comment:
this isnt type-preserving?

Reply:
see above

Reply:
wow i understood typing less than i thought.  thanks

Reply:
this is because we `raise AbstractMethodError(self)` here and all the subclasses have the signature defined to return same type as self.